testing/run_tests.py

```diff
@@ -605,25 +605,53 @@ def RunObjcTests(ios_variant='ios_debug_sim_unopt', test_filter=None):
   ios_out_dir = os.path.join(out_dir, ios_variant)
   EnsureIosTestsAreBuilt(ios_out_dir)
 
-  ios_unit_test_dir = os.path.join(
-      buildroot_dir, 'flutter', 'testing', 'ios', 'IosUnitTests'
-  )
+  new_simulator_name = 'IosUnitTestsSimulator'
 
-  # Avoid using xcpretty unless the following can be addressed:
-  # - Make sure all relevant failure output is printed on a failure.
-  # - Make sure that a failing exit code is set for CI.
-  # See https://github.com/flutter/flutter/issues/63742
-  command = [
-      'xcodebuild '
-      '-sdk iphonesimulator '
-      '-scheme IosUnitTests '
-      "-destination platform='iOS Simulator,name=iPhone 11' "
-      'test '
-      'FLUTTER_ENGINE=' + ios_variant
+  # Delete simulators with this name in case any were leaked
+  # from another test run.
+  DeleteSimulator(new_simulator_name)
```

Comment:
Another option is to detect if the simulator exists and decide to create a new one if one does not exist.

Reply:
That seemed like a lot more work and parsing 🙂 

I wanted to guard against the simulator leaking when this script is run multiple times for flake detection.  Fortunately creating and deleting simulators is extremely fast, so we don't need to over optimize.
```
xcrun simctl create test1 com.apple.CoreSimulator.SimDeviceType.iPhone-11  0.07s user 0.07s system 65% cpu 0.217 total
```
```
xcrun simctl delete test1  0.08s user 0.08s system 86% cpu 0.180 total
```